pkg/gitignore/gitignore_test.go

```diff
@@ -57,6 +59,9 @@ func TestGitignoreMatch(t *testing.T) {
 		{Name: "Normalize globs, One Path", Path: "ab/cd/fg/e.txt", Pattern: "a**b/c***d/**/e.txt", Expected: true},
 		{Name: "Normalize globs, Two Paths", Path: "ab/cd/fg/hi/e.txt", Pattern: "a**b/c***d/**/e.txt", Expected: true},
 		{Name: "Normalize globs, Zero Paths", Path: "ab/cd/e.txt", Pattern: "a**b/c***d/**/e.txt", Expected: true},
+		{Name: "Negated globs", Path: "a/b/c.md", Pattern: "!**.md", Expected: false},
```

Comment:
if you open a PR with just these tests, i'll be happy to accept, im pretty sure it will pass as is